### PR TITLE
Fix warnings in CLI destroy tests

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**Problem**
See #6450 which discusses a warning for the destroy tests related to cells.

I was able to replicate the issue locally when restricting the number of threads used within the test:
```
yarn jest src/commands/destroy/cell/__tests__/cell.test.js --maxWorkers=1
```

**Solution**
Ensure async functions are awaited.

**Changes**
 1. Removed file mocking within `beforeEach` as it's not needed before each test given the second test needs different files mock.
 2. `await` all `files(...)` calls
 3. `await` listr task run before expecting tests

**Questions**
 1. Other tests e.g. page seem to also not await `files` - this is also an issue?
 2. Happy with the async/await approach over then's?

### CODE REVIEW

**Comment by dac09** [comment]
Thanks @Josh-Walker-GM - found this while reviewing your PR - much appreciated!

File: packages/cli/src/commands/destroy/cell/__tests__/cell.test.js [hunk 1]
```diff
@@ -23,16 +23,15 @@ import '../../../../lib/test'
 import { files } from '../../../generate/cell/cell'
 import { tasks } from '../cell'
 
-beforeEach(() => {
-  fs.__setMockFiles(files({ name: 'User' }))
-})
+beforeEach(() => {})
```
Comment (dac09) [inline packages/cli/src/commands/destroy/cell/__tests__/cell.test.js:26]:
```suggestion
```

**Comment by dac09** [comment]
> Other tests e.g. page seem to also no await files - this is also an issue?

Yes ideally we can move this over to the syntax you're using! I tried this earlier today but didn't have enough time to debug why my changes made the tests fail! 

> Happy with the async/await approach over then's?

Definitely 💫

**Comment by Josh-Walker-GM** [comment]
> Thanks @Josh-Walker-GM - found this while reviewing your PR - much appreciated!

Thought I should have a go fixing the issue given my changes highlighted/caused it 😆

**Comment by Josh-Walker-GM** [comment]
If you wish I can take a look at updating the remaining tests so they await the files and other async calls? This is best in another PR but is it okay if I update all the remaining tests within that one PR?

**Comment by Josh-Walker-GM** [comment]
> > Other tests e.g. page seem to also no await files - this is also an issue?
> 
> Yes ideally we can move this over to the syntax you're using! I tried this earlier today but didn't have enough time to debug why my changes made the tests fail!

In theory this could lead to tests passing thought the correct behaviour, no?

**Comment by Josh-Walker-GM** [comment]
Very happy to update the rest within this PR.

I hopefully won't run into any tests which currently pass due to the async but in theory shouldn't when sync'd. If I run into any difficulties that I can't solve I'll update here too if that's fine?

**Comment by dac09** [comment]
Yes ofcourse, that’s great. I think the other destroy commands don’t have the async problem anyway, so should be straightforward!

**Comment by Josh-Walker-GM** [comment]
I did a pass through the tests and a few local runs and didn't notice any other issues with the tests. I removed an unneeded `async` but that's it.

**Review by dac09 — APPROVED** [review]
Thank you again Josh, on 🔥

**Pull request merged**